core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,423 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+   // With "GetSignature" we get the arguments of "Method" and put them in a TString.
+   // Then the TString "MethodArguments" is modified.
+   MethodArguments.ReplaceAll(" = ", "=");
+   // maybe add also " =" and "= ";
+   MethodArguments.ReplaceAll("* ", " *");
+   // is this always? below
+   MethodArguments.ReplaceAll("*=", " *=");
+   MethodArguments.ReplaceAll("*)", " *)");
+   MethodArguments.ReplaceAll("*,", " *,");
+   MethodArguments.ReplaceAll("*& ", " *&");
+   MethodArguments.ReplaceAll("& ", " &");
+   MethodArguments.ReplaceAll("ostream", "std::ostream");
+   MethodArguments.ReplaceAll("istream", "std::istream");
+   MethodArguments.ReplaceAll("*msgfmt", "*msgfmt,...");
+   TString Buff ("\\b");
+   Buff.Append(ClassName);
+   Buff.Append("::\\b");
+   TPRegexp ArgFix(Buff.Data());
+   ArgFix.Substitute(MethodArguments, "");
+   return MethodArguments;
+}
+}
+
+namespace {
+static TString 
+InlineFuncCheck(TString ClassName, TString Method)
+{
+   // The function checks if a member function (Method) of a class (ClassName) is
+   // defined as inline. If so, it also checks if it is virtual.
+   // Then the return type of "Method" is changed for the need of Doxygen and with
+   // respect to the function definition. The corrected "ReturnType" is returned.
+
+   Int_t InlineCheck = (((TMethod *)TClass::GetClass(ClassName)->GetListOfMethods()->FindObject(Method))->ExtraProperty() & kIsInlined);
+   // We put the return type of "Method" in a TString "ReturnTypeName".
+   TString ReturnTypeName(((TMethod *)TClass::GetClass(ClassName)->GetListOfMethods()->FindObject(Method))->GetReturnTypeName());
+   TString Buff ("\\b");
+   Buff.Append(ClassName);
+   Buff.Append("::\\b");
+   TPRegexp ReturnFix(Buff.Data());
+   ReturnFix.Substitute(ReturnTypeName, "");
+   // We define the TString witch will be modified and returned from "InlineFunctionCheck".
+   TString ReturnType("");
+   // InlineCheck is not zero only if the function is defined as inline.
+   if (InlineCheck == 0) {
+      ReturnType.Append(ReturnTypeName);
+      // If the function is not inline we only change the spacing in "ReturnType".
+      ReturnType.ReplaceAll("*", " *");
+   } else {
+      // We check if the function is defined as virtual.
+      Int_t VirtualCheck = (((TMethod *)TClass::GetClass(ClassName)->GetListOfMethods()->FindObject(Method))->Property() & kIsVirtual);
+      if (VirtualCheck != 0) {
+         // If the function is virtual, we append "virtual" before the return type.
+         TString Buffer("virtual ");
+         Buffer.Append(ReturnTypeName);
+         ReturnType.Append(Buffer);
+      } else {
+         // If it's not virtual we only change the spacing in "ReturnType".
+         ReturnType.Append(ReturnTypeName);
+         ReturnType.ReplaceAll(" *", "*");
+      }
+   }
+   // In any case we need to change the return type as following.
+   ReturnType.ReplaceAll("istream", "std::istream");
```

Comment:
Don't we have the same issue with `vector`, `shared_ptr` etc - any class in `std`?